### ISSUE TITLE
OsmAnd deletes favorites if one is saved before all favorites are loaded

### ISSUE THREAD

**Issue by burillo-se** [osmandapp/OsmAnd#20957]
### Description

If OsmAnd hasn't loaded its favorites yet, and I create a new one, the old ones are deleted and there appears to be no way to restore them.

(apologies, there was a rant previously, but backups work, it was user error)

### Steps to reproduce

1. Have lots of favorites
2. Open OsmAnd app (the favorites will not be loaded until some time later - maybe this is specific to having favorite on SD card)
3. Save a favorite

### Actual result

All old favorites are deleted

### Expected result

If favorites are unavailable right now, I'd expect either that I wouldn't be able to save a new one, or, preferably, it'll be saved and then added to favorites later.

### Your Environment (required)

**WARNING** Crash-Logs **MAY** contain information you deem sensitive.
Review this **CAREFULLY** before posting your issue!

```
OsmAnd Version: plus, 4.8.6
Android/iOS version: 12
Device model: Nokia 8.3
Crash-Logs: ?
```

**Comment by yuriiurshuliak** [comment]
Unfortunately, I do not fully understand the steps for reproduction. Am I correct in assuming that after opening the app with a large number of favorites, it takes some time for the favorites to load, and during that period, a new favorite needs to be created?

Could you please provide a video and more detailed steps to help us reproduce the bug accurately without making assumptions?

**Comment by anatolyburakov** [comment]
Yes, your understanding is correct: when I first open OsmAnd, I have no favorites, but they appear after a while (usually about a minute in). During that period, if I save a favorite, all previous favorites that I had get deleted. I'm not sure if this is specific to having OsmAnd data stored on SD card (as opposed to in main memory), but I'm mentioning it just in case it is relevant.

Here's a link to GDrive with a video showing the period between loading OsmAnd and appearance of favoties: https://drive.google.com/file/d/1Csb3Km9FxkvviNMIlo267PhGXu_I7dem/view

If I save a favorite during this period, other favorites are deleted (as in, even after waiting for a few minutes, the only favorites I will see is the one I just created, and if I reload OsmAnd I will see the same - just one favorite).

**Comment by yuriiurshuliak** [comment]
The bug has been successfully reproduced. To reproduce it, import a large number of favorites, fully close the application, then reopen it and go to My Places > Favorites. You’ll notice that not all the favorites load right after opening the app. Close the My Places tab, create a new favorite on the map, and close the app again. Upon reopening and navigating back to My Places > Favorites, only the newly created favorite will appear, while the previously imported favorites will no longer be visible.

OsmAnd~ 4.9.0#3340m, released: 2024-10-02

<table style="width:100%">
  </tr>
  <tr>
    <td> <video source src="https://github.com/user-attachments/assets/cdf06286-64a1-47f3-8f55-5e675b1a716b"  width="300" height="auto"/>   </td>
<th><img src="https://github.com/user-attachments/assets/2a6a3b55-d658-43de-b164-494614cd240c"  width="300" height="auto"/> </th>
  </tr>
</table>

**Comment by sonora** [comment]
From my testing, the behavior even seems somehow _triggered_ by going to the My Favorites screen too soon after app start:
- If I immediately go there after the map screen appears after a new app start, I get this 1+ minute or so "dead zone" where the app becomes unresponsive, sometimes the OS even warns about it
- If instead I wait for ten seconds after the map screen has appeared before I go there, all is fine: Favorites are there, all is responsive, no hanging.

So while the 10 seconds may vary with your number of favorites, it seems is aggravated by going to some screen action too early in the process, creating this extra-long blocking. I believe I have seen similar effects by trying certain search operations too early after app start.
